src/app/abstraction-field/abstraction-field.component.ts

```diff
@@ -283,7 +289,8 @@ export class AbstractionFieldComponent implements OnInit {
           user: this.role.userMail(),
           fieldId: field.medicalRecordAbstractionFieldId,
           nameValues: [ {name: this.activityOfField + "_" + fieldName, value: v},
-            {name: this.activityOfField + "_fileName", value: field.fieldValue.fileName} ]
+            {name: this.activityOfField + "_fileName", value: field.fieldValue.fileName} ],
+          realm : localStorage.getItem( ComponentService.MENU_SELECTED_REALM )
```

Comment:
Instead of putting the same line `realm : localStorage.getItem( ComponentService.MENU_SELECTED_REALM )` in every if block, maybe it is better to put it on line 301 `patch['realm'] = localStorage.getItem( ComponentService.MENU_SELECTED_REALM )` or on line 199 where it is declared, since `realm` is necessary for every kind of condition, you can initialize it already with realm on that line?